ros_ws/src/dataset_generation/include/dataset_generation/filter_point_clouds.h

```diff
@@ -26,40 +26,33 @@ namespace dataset_generation
 class PointCloudFiltering
 {
 public:
-    PointCloudFiltering(
-        std::string in_folder_path,
-        std::string out_folder_path);
+    PointCloudFiltering();
 
     ~PointCloudFiltering();
 
-    void apply_voxel_filter_and_save(float leaf_size_x, float leaf_size_y, float leaf_size_z);
-    void apply_voxel_filter_and_visualize(float leaf_size_x, float leaf_size_y, float leaf_size_z);
-private:
-    std::string in_folder_path_;
-    std::string out_folder_path_;
+    void apply_voxel_filter_and_save(std::string in_folder_path,
+        std::string out_folder_path,float leaf_size_x, float leaf_size_y, float leaf_size_z);
+    void apply_voxel_filter_and_visualize(std::string in_folder_path,
+        std::string out_folder_path,float leaf_size_x, float leaf_size_y, float leaf_size_z);
 };
-PointCloudFiltering::PointCloudFiltering(
-        std::string in_folder_path,
-        std::string out_folder_path)
-{
-    in_folder_path_=in_folder_path;
-    out_folder_path_=out_folder_path;
-}
+PointCloudFiltering::PointCloudFiltering()
+{}
 PointCloudFiltering::~PointCloudFiltering()
 {}
 
 /*
 apply_voxel_filter_and_save method is a modified version of the VoxelGrid filtering found at: 
 https://pcl-tutorials.readthedocs.io/en/master/voxel_grid.html?highlight=voxelGrid
 */
-void PointCloudFiltering::apply_voxel_filter_and_visualize(float leaf_size_x, float leaf_size_y, float leaf_size_z)
+void PointCloudFiltering::apply_voxel_filter_and_visualize(std::string in_folder_path,
+            std::string out_folder_path,float leaf_size_x, float leaf_size_y, float leaf_size_z)
```

Comment:
The `in_folder_path` and `out_folder_path` are misleading, can you rename them to `in_pcd_path` and `out_pcd_path`? Since there are single pcd files and not folders? You'll have to change this throughout this file.